irohad/network/sync_gate.hpp

```diff
@@ -0,0 +1,24 @@
+//
+// Created by Takumi Yamashita on 2017/07/11.
+//
+
+#ifndef IROHA_NETWORK_SYNCGATE_HPP
+#define IROHA_NETWORK_SYNCGATE_HPP
+namespace iroha {
+  namespace network {
+
+    /**
+     * Interface for downloading blocks from a network
+     */
+    class SyncGate {
```

Comment:
We already have such interface. It is called BlockLoader. https://github.com/hyperledger/iroha/blob/develop/irohad/network/block_loader.hpp